src/coreclr/jit/codegenxarch.cpp

```diff
@@ -1979,9 +1982,32 @@ void CodeGen::genMultiRegStoreToSIMDLocal(GenTreeLclVar* lclNode)
         inst_RV_RV_IV(INS_shufpd, EA_16BYTE, targetReg, targetReg, 0x01);
     }
     genProduceReg(lclNode);
-#else  // !UNIX_AMD64_ABI
-    assert(!"Multireg store to SIMD reg not supported on Windows");
-#endif // !UNIX_AMD64_ABI
+#elif defined(TARGET_X86) && defined(TARGET_WINDOWS)
+    assert(varTypeIsIntegral(retTypeDesc->GetReturnRegType(0)));
+    assert(varTypeIsIntegral(retTypeDesc->GetReturnRegType(1)));
+    assert(lclNode->TypeIs(TYP_SIMD8));
+
+    // This is a case where a SIMD8 struct returned as [EAX, EDX]
+    // and needs to be assembled into a single xmm register,
+    // note we can't check reg0=EAX, reg1=EDX because they could be already moved.
+
+    inst_RV_RV(ins_Copy(targetReg, TYP_FLOAT), targetReg, reg0, TYP_INT);
+    if (compiler->compOpportunisticallyDependsOn(InstructionSet_SSE41))
+    {
+        emitAttr size = emitTypeSize(TYP_SIMD8);
+        GetEmitter()->emitIns_R_R_I(INS_pinsrd, size, targetReg, reg1, 1);
+    }
+    else
+    {
+        regNumber tempXmm = lclNode->GetSingleTempReg();
+        inst_RV_RV(ins_Copy(tempXmm, TYP_FLOAT), tempXmm, reg1, TYP_INT);
+        inst_RV_RV(INS_punpckldq, targetReg, tempXmm, TYP_SIMD8);
+    }
+
+    inst_RV_RV(ins_Copy(targetReg, TYP_FLOAT), targetReg, reg1, TYP_INT);
```

Comment:
As per https://github.com/dotnet/runtime/pull/52581/files#r630238514, this is meant to be used as inst_RV_RV(ins_Copy(srcReg, tgtType), tgtReg, srcReg, srcType) and is going to pick movaps rather than movd with how its currently being used.

However, its also not clear what this is needed for? I thought this was for going from `[EAX, EDX]` to a `XMM` register, in which case this is overwriting element 0?